doc/devices/qsim.rst

```diff
@@ -0,0 +1,113 @@
+The qsim device
+===============
+
+qsim is a Schrödinger full state-vector simulator while qsimh is a hybrid
+Schrödinger-Feynman simulator, cutting the qubit lattice into two parts and then
+using qsim to simulate each part individually. For further details see the `qsim
+website <https://github.com/quantumlib/qsim>`__. These simulators are
+considerably faster compared to the Cirq simulator for many qubits, as can be
+seen in the benchmark below.
+
+.. image:: ../_static/qsim_bench.png
+    :align: center
+    :width: 70%
+
+You can instantiate the qsim device in
+PennyLane as follows:
+
+.. code-block:: python
+
+    import pennylane as qml
+
+    dev = qml.device('cirq.qsim', wires=2)
+
+This device can then be used just like other devices for the definition and evaluation of QNodes within PennyLane.
+A simple quantum function that returns the expectation value of a measurement and depends on three classical input
+parameters would look like:
+
+.. code-block:: python
+
+    @qml.qnode(dev)
+    def circuit(x, y, z):
+        qml.RZ(z, wires=[0])
+        qml.RY(y, wires=[0])
+        qml.RX(x, wires=[0])
+        qml.CNOT(wires=[0, 1])
+        return qml.expval(qml.PauliZ(wires=1))
+
+You can then execute the circuit like any other function to get the quantum mechanical expectation value.
+
+.. code-block:: python
+
+    circuit(0.2, 0.1, 0.3)
+
+Device options
+~~~~~~~~~~~~~~
+
+qsim and qsimh use the same way of defining qubits as Cirq, e.g., ``LineQubit``
+or ``GridQubit``. As with the Cirq device, the qsim and qsimh devices therefore
+accept an additional argument ``qubits=None`` that can be used to define your
+own qubits and give them to the device as a list.
+
+.. code-block:: python
+
+    import cirq
+
+    qubits = [
+      cirq.GridQubit(0, 0),
+      cirq.GridQubit(0, 1),
+      cirq.GridQubit(1, 0),
+      cirq.GridQubit(1, 1),
+    ]
+
+    dev = qml.device("cirq.qsim", wires=4, qubits=qubits)
+
+The wire of each qubit corresponds to its index in the ``qubit`` list. In the above example,
+the wire 2 corresponds to ``cirq.GridQubit(1, 0)``.
+
+If no qubits are given, the plugin will create an array of ``LineQubit`` instances.
+
+qsimh options
+^^^^^^^^^^^^^
+
+qsimh requires specific options to be set for the simulator. These can be passed
+by the positional argument ``qsimh_options``. See the `QSimh usage documentation
```

Comment:
I've been super-confused whether to spell it QSim or qsim (since I've seen both in different places). They use qsim and qsimh on the website and in the supremacy paper, so I went with that. :laughing: